tracer/src/Datadog.Trace/Debugger/Snapshots/DebuggerSnapshotSerializer.cs

```diff
@@ -18,12 +18,19 @@ internal static class DebuggerSnapshotSerializer
         private const string ReachedTimeoutMessage = "Reached timeout";
         private const string ReachedNumberOfObjectsMessage = "Reached the maximum number of objects";
         private const string ReachedNumberOfItemsMessage = "Reached the maximum number of items";
-        private const int MaximumNumberOfItemsInCollectionToCopy = 100;
-        private const int MaximumNumberOfFieldsToCopy = 1000;
+
         private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(DebuggerSnapshotSerializer));
-        private static readonly ImmutableDebuggerSettings DebuggerSettings = ImmutableDebuggerSettings.Create(Debugger.DebuggerSettings.FromDefaultSource());
-        private static readonly int MaximumDepthOfMembersToCopy = DebuggerSettings.MaximumDepthOfMembersOfMembersToCopy;
-        private static readonly int MillisecondsToCancel = DebuggerSettings.MaxSerializationTimeInMilliseconds;
+
+        private static int _maximumNumberOfItemsInCollectionToCopy = DebuggerSettings.DefaultMaxNumberOfItemsInCollectionToCopy;
+        private static int _maximumNumberOfFieldsToCopy = DebuggerSettings.DefaultMaxNumberOfFieldsToCopy;
+        private static int _maximumDepthOfMembersToCopy = DebuggerSettings.DefaultMaxNumberOfFieldsToCopy;
+        private static int _maximumSerializationTime = DebuggerSettings.DefaultMaxSerializationTimeInMilliseconds;
+
```

Comment:
Do we actually need these? Why can't we just inline all of them (these are `const`s so the time in which they are evaluated is meaningless)?

Reply:
We don't need all of them, but only a part, which are settable from Env variables.
I like to be consistent, so if we setting any of them from debugger settings, I prefer to set all of them.

Reply:
That's fair, resolving.